### PR TITLE
fix: update return type of function exported by default (#273)

### PR DESCRIPTION
**Pull request by tearwyx**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -121,8 +127,9 @@ interface PrettyOptions_ {
 declare namespace PinoPretty {
   type Prettifier = (inputData: string | object) => string;
   type MessageFormatFunc = (log: LogDescriptor, messageKey: string, levelLabel: string) => string;
-  type PrettyOptions = PrettyOptions_
+  type PrettyOptions = PrettyOptions_;
+  type PrettyStream = Transform & OnUnknown;
```
Comment (kibertoad) [inline index.d.ts:131]: Should it be both and not either?

File: test/types/pino-pretty.test-d.ts [hunk 2]
```diff
@@ -52,16 +52,11 @@ const options2: PrettyOptions = {
...
+expectType<PrettyStream>(pretty(options));
+expectType<PrettyStream>(PinoPrettyNamed(options));
+expectType<PrettyStream>(PinoPrettyDefault(options));
+expectType<PrettyStream>(PinoPrettyStar.PinoPretty(options));
+expectType<PrettyStream>(PinoPrettyStar.default(options));
+expectType<PrettyStream>(PinoPrettyCjsImport.PinoPretty(options));
+expectType<PrettyStream>(PinoPrettyCjsImport.default(options));
 expectType<any>(PinoPrettyCjs(options));
-
-expectType<string>(pretty({ foo: "bar" }));
```
Comment (kibertoad) [inline test/types/pino-pretty.test-d.ts:66]: why were these removed?

Comment (tearwyx) [reply test/types/pino-pretty.test-d.ts:66]:
```
This expression is not callable.
  Type 'Transform & OnUnknown' has no call signatures. ts(2349)
```

File: index.d.ts (hunk 1 above)
Comment (tearwyx) [reply index.d.ts:131]: If you mean intersection operands, then function `pino-pretty` returns the return value of function `pino-abstract-transport` and this return value is of this type.

**Review by kibertoad — APPROVED** [review]

**Comment by mcollina** [comment]
cc @kibertoad

**Review by mcollina — APPROVED** [review]
lgtm
